find.go

```diff
@@ -90,14 +94,56 @@ func (s *server) doFind(ctx context.Context, method, source string, req *url.URL
 			return nil, err
 		}
 
+		// checking whether it *might* be double hashed query, i.e.
+		// - either decoded multihash from the path has DBL_SHA2_256 codec
+		// - or it's a POST batch reqest (those do not get submitted often)
+		mightBeDhash := false
+		if req.Method == "GET" {
+			ss := strings.Split(b.Path, "/")
+			// multihash is the last part in the path
+			mh, err := multihash.FromB58String(ss[len(ss)-1])
+			if err == nil {
+				dmh, err := multihash.Decode(mh)
+				mightBeDhash = err == nil && dmh.Code == multihash.DBL_SHA2_256
+			}
+		} else {
+			mightBeDhash = true
+		}
+
 		switch resp.StatusCode {
 		case http.StatusOK:
 			atomic.AddInt32(&count, 1)
 			providers, err := model.UnmarshalFindResponse(data)
-			if err != nil {
-				return nil, err
+			// if there was no error then it was a regular find response
+			if err == nil {
+				return &providers, nil
+			} else {
+				// if there was an eror and the query isn't double hashed - then just return the error
+				if !mightBeDhash {
+					return nil, err
+				}
+				// otherwise verifying whether that is a response from dhstore
+				lookupResponse := &dhstore.LookupResponse{}
+				err := json.Unmarshal(data, lookupResponse)
+				if err != nil {
+					return nil, err
+				}
+				// convert dhstore response to find response
+				findResponse := &model.FindResponse{
+					EncMultihashResults: make([]model.EncMultihashResult, 0, len(lookupResponse.EncryptedMultihashResults)),
```

Comment:
We shlould either rename dhstore.LookupResponse to use `EncMultihashResults` or storetheindex to use `EncryptedMultihashResults`. 

My vote is the latter. That removes the need for conversion here and scatter gather would just work without needing to fiddle with JSON.